tests/functional/unit_testing/test_ut_adapter_hooks.py

```diff
@@ -0,0 +1,50 @@
+from unittest import mock
+
+import pytest
+
+from dbt.tests.util import run_dbt, run_dbt_and_capture
+from dbt_common.exceptions import CompilationError
+from tests.functional.unit_testing.fixtures import (
+    my_model_a_sql,
+    my_model_b_sql,
+    my_model_sql,
+    test_my_model_pass_yml,
+)
+
+
+class BaseUnitTestAdapterHook:
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "my_model.sql": my_model_sql,
+            "my_model_a.sql": my_model_a_sql,
+            "my_model_b.sql": my_model_b_sql,
+            "test_my_model.yml": test_my_model_pass_yml,
+        }
+
+
+class TestUnitTestAdapterHookPasses(BaseUnitTestAdapterHook):
+    def test_unit_test_runs_adapter_pre_hook(self, project):
+        results = run_dbt(["run"])
+        assert len(results) == 3
+
+        mock_pre_model_hook = mock.Mock()
+        with mock.patch.object(type(project.adapter), "pre_model_hook", mock_pre_model_hook):
+            results = run_dbt(["test", "--select", "test_name:test_my_model"], expect_pass=True)
+
+            assert len(results) == 1
+            mock_pre_model_hook.assert_called_once()
+
+
+class TestUnitTestAdapterHookFails(BaseUnitTestAdapterHook):
```

Comment:
Do we feel we also need to have tests for `post_model_hook`?

Reply:
I think it's worth it since they are different codepaths and could regress for different reasons.

Reply:
Will update